scipy/diff/statsmodels/tests/test__derivative.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import division, absolute_import
+import numpy as np
+from numpy.testing import (assert_, run_module_suite)
+from scipy.diff.statmodels._derivative import derivative, gradient, jacobian
+
+
+def fun(x):
+    return x**2
+
+
+def df_fun(x):
+    return 2*x
```

Comment:
To be able to easily extend the set of functions tested, I would do something like:

    funcs = [
        (lambda x: x**2, lambda x: 2*x),
        (lambda x: np.exp(x), lambda x: np.exp(x)),
    ]

And then in the test loop over ``funcs``. If needed, add a tolerance per function as the third tuple element